homeassistant/components/firmata/board.py

```diff
@@ -0,0 +1,196 @@
+
+    board_data['port_discovery_exceptions'] = True
+
+    board = PymataCore(**board_data)
+
+    await board.start_aio()
+    return board
+
+
+class FirmataBoardPin(Entity):
+    """Manages a single Firmata board pin."""
+
+    def __init__(self, hass, board_name: str, **kwargs):
+        """Initialize the pin."""
+        self.hass = hass
+        self._name = kwargs[CONF_NAME]
+        self._state = None
+        self._board_name = board_name
+        self._board = hass.data[DOMAIN][self._board_name]
+        self._conf = kwargs
+        _LOGGER.info(self._conf)
+        self._pin = self._conf[CONF_PIN]
+        self._pin_mode = self._conf[CONF_PIN_MODE]
+        self._firmata_pin_mode = None
+        self._firmata_pin = self._pin
+        if isinstance(self._pin, str):
+            self._pin_type = 'analog'
+            self._firmata_pin = int(self._firmata_pin[1:])
+            self._firmata_pin += self._board.api.first_analog_pin
+        else:
+            self._pin_type = 'digital'
+        self._location = (DOMAIN, self._board_name, 'pin', self._pin)
+        self._unique_id = '_'.join(str(i) for i in self._location)
+        self._identifiers = {
+            (DOMAIN, self._unique_id)
+        }
+        self._attributes = {}
+
+    def _set_attributes(self):
+        """Set the entity's attributes."""
+        self._attributes = {
+            'config_entry_id': self._board.config_entry.entry_id,
+            'via_hub': (DOMAIN, self._board_name),
+            'pin_type': self._pin_type,
+            'firmata_pin': self._firmata_pin,
+            'firmata_pin_mode': self._firmata_pin_mode,
+            'identifiers': self._identifiers,
+            'manufacturer': 'Firmata'
+        }
+        self._attributes.update(self._conf)
+
+    def _mark_pin_used(self):
+        """Test if a pin is used already on the board or mark as used."""
+        if self._location in self._board.used_pins:
+            return False
+        self._board.used_pins.append(self._location)
+        return True
+
+    @property
+    def name(self) -> str:
+        """Get the name of the pin."""
+        return self._name
+
+    @property
+    def should_poll(self) -> bool:
+        """No polling needed."""
+        return False
+
+    @property
+    def available(self) -> bool:
+        """Return True because the board is always available."""
+        return True
+
+    @property
+    def device_state_attributes(self) -> dict:
+        """Return device specific state attributes."""
+        if self._state is not None:
+            self._attributes['state'] = self._state
+        return self._attributes
+
+    @property
+    def unique_id(self):
+        """Return a unique identifier for this device."""
+        return self._unique_id
+
+    @property
+    def device_info(self) -> dict:
+        """Return device info."""
+        device_info = {
+            'name': self.name
+        }
+        device_info.update(self._attributes)
```

Comment:
I see! I'll put all of the pin info in the device info only then.

Reply:
OK, so I understand now that device info is for static information about the device and that device state attributes and entity state attributes are for dynamic values. However, what is the difference between device state attributes and entity state attributes?

Reply:
There's nothing called `entity state attributes`. The property on the entity is called `device_state_attributes`. This will be used to populate the state attributes for the entity.

The naming `device_state_attributes` has stuck although it would make more sense to name it `entity_state_attributes`.